test/index.ts

```diff
@@ -299,77 +278,74 @@ const testTable: Partial<AbstractSqlModel> = {
 };
 
 test(
-	'correct read types for a test table',
-	testTable,
-	source`
-		export interface Parent {
-			created_at: DateString;
-			modified_at: DateString;
-			id: number;
-		}
-
-		export interface Other {
-			created_at: DateString;
-			modified_at: DateString;
-			id: number;
-			is_referenced_by__test?: Test[];
-		}
-
-		export interface Test {
-			created_at: DateString;
-			modified_at: DateString;
-			id: number;
-			a_date: DateString;
-			a_file: WebResource;
-			parent: { __id: Parent['id'] } | [Parent];
-			references__other: { __id: Other['id'] } | [Other];
-			test__has__tag_key?: TestTag[];
-			test_tag?: TestTag[];
-		}
-
-		export interface TestTag {
-			created_at: DateString;
-			modified_at: DateString;
-			test: { __id: Test['id'] } | [Test];
-			tag_key: string;
-			id: number;
-		}
-	`,
-);
-
-test(
-	'correct write types for a test table',
+	'correct types for a test table',
 	testTable,
 	source`
 		export interface Parent {
-			created_at: Date;
-			modified_at: Date;
-			id: number;
+			Read: {
+				created_at: Types['Date Time']['Read'];
+				modified_at: Types['Date Time']['Read'];
+				id: Types['Serial']['Read'];
+			}
+			Write: {
+				created_at: Types['Date Time']['Write'];
+				modified_at: Types['Date Time']['Write'];
+				id: Types['Serial']['Write'];
+			}
 		}
 
 		export interface Other {
-			created_at: Date;
-			modified_at: Date;
-			id: number;
+			Read: {
+				created_at: Types['Date Time']['Read'];
+				modified_at: Types['Date Time']['Read'];
+				id: Types['Serial']['Read'];
+				is_referenced_by__test?: Test['Read'][];
+			}
+			Write: {
+				created_at: Types['Date Time']['Write'];
+				modified_at: Types['Date Time']['Write'];
+				id: Types['Serial']['Write'];
+			}
 		}
 
 		export interface Test {
-			created_at: Date;
-			modified_at: Date;
-			id: number;
-			a_date: Date;
-			a_file: WebResource;
-			parent: Parent['id'];
-			references__other: Other['id'];
+			Read: {
+				created_at: Types['Date Time']['Read'];
+				modified_at: Types['Date Time']['Read'];
+				id: Types['Serial']['Read'];
+				a_date: Types['Date']['Read'];
+				a_file: Types['WebResource']['Read'];
+				parent: { __id: Parent['Read']['id'] } | [Parent['Read']];
+				references__other: { __id: Other['Read']['id'] } | [Other['Read']];
+				test__has__tag_key?: TestTag['Read'][];
+				test_tag?: TestTag['Read'][];
```

Comment:
maybe this could be `Array<TestTag['Read']>`  instead of `TestTag['Read'][]` to match with most modern linters will generate too

Reply:
Done